### PR TITLE
Save profile images directly to s3

### PR DESCRIPTION
**Pull request by c-o-l-o-r**
@villanuevawill @codeluggage this is the API portion of this task

### CODE REVIEW

File: bounties_api/user/views.py [hunk 1]
```diff
@@ -47,6 +51,46 @@ def get(self, request):
...
+        current_user = request.current_user
+        nonce = int(random() * 1000)
+        bucket = 'assets.bounties.network'
+
+        sm_key = '{}/userimages/{}-sm-{}.png'.format(settings.ENVIRONMENT, current_user.public_address, nonce)
+        lg_key = '{}/userimages/{}-lg-{}.png'.format(settings.ENVIRONMENT, current_user.public_address, nonce)
+
+        sm_put_url = client.generate_presigned_url(
+            'put_object',
+            Params={
```
Comment (villanuevawill) [inline bounties_api/user/views.py]: @c-o-l-o-r looking good - you just need to set the `'CacheControl': 'max-age=31536000'`

File: bounties_api/user/views.py [hunk 2]
```diff
@@ -47,6 +51,46 @@ def get(self, request):
...
+                'ContentType': 'image/png',
+                'Bucket': bucket,
+                'ACL': 'public-read',
+                'Key': sm_key}
```
Comment (villanuevawill) [inline bounties_api/user/views.py]: Also - can you look up the thing I was telling you. There is a way to make it so the name of the file when you download it keeps its original name (without the nonce) although the key and url has the nonce.

File: bounties_api/user/views.py (hunk 1 above)
Comment (c-o-l-o-r) [reply bounties_api/user/views.py]: i added this on the frontend!

File: bounties_api/user/views.py (hunk 2 above)
Comment (c-o-l-o-r) [reply bounties_api/user/views.py]: i also added this on the frontend!

File: bounties_api/user/views.py [hunk 3]
```diff
@@ -47,6 +51,46 @@ def get(self, request):
...
 
+class RequestProfileImageUploadURL(APIView):
+    permission_classes = [AuthenticationPermission]
+
+    def get(self, request):
+        AWS_REGION = 'us-east-1'
+        client = boto3.client('s3', region_name=AWS_REGION)
+
+        current_user = request.current_user
+        nonce = int(random() * 1000)
```
Comment (villanuevawill) [inline bounties_api/user/views.py]: this works... but there is also a uuid.uuidv4 library I use elsewhere :)

**Comment by villanuevawill** [comment]
Do we need to do resizing on the github images?

File: bounties_api/user/views.py (hunk 3 above)
Comment (c-o-l-o-r) [reply bounties_api/user/views.py]: good point, i switched it out :)

**Comment by c-o-l-o-r** [comment]
> Do we need to do resizing on the github images?

Yeah we should. Fortunately github will return whatever size you ask for in the get request, so I just make two requests: one for the small size and one for the large.

**Pull request merged**